### PR TITLE
MESH-1592: balance_at RPC

### PR DESCRIPTION
**Pull request by vkomenda**
### new features

- Added a new `asset_balanceAt` RPC call to query identity-ticker balance at a checkpoint.

### CODE REVIEW

**Review by maxsam4 — COMMENTED** [review]
The polymesh schema file needs to be updated

File: polymesh_schema.json [hunk 1]
```diff
@@ -1525,6 +1525,27 @@
...
+                        "type": "Ticker",
+                        "isOptional": false
+                    },
+                    {
+                        "name": "checkpoint",
+                        "type": "CheckpointId",
+                        "isOptional": false
+                    }
+                ],
+                "type": "BalanceAtResult<Balance>"
```
Comment (maxsam4) [inline polymesh_schema.json:1548]: You need to also define what `BalanceAtResult` is. 
Also, we can't really have generic type params for custom structs in the schema so you'd need to define `BalanceAtResult` as a fixed type. See `CanTransferResult` for example.

**Review by Centril — APPROVED** [review]
Aside from @maxsam4's comments, this LGTM

**Review by maxsam4 — APPROVED** [review]

File: pallets/runtime/common/src/runtime.rs [hunk 2]
```diff
@@ -845,6 +845,19 @@ macro_rules! runtime_apis {
...
                 }
+
+                #[inline]
+                fn balance_at(
+                    ticker: Ticker, checkpoint: CheckpointId, dids: Vec<IdentityId>
+                ) -> node_rpc_runtime_api::asset::BalanceAtResult
+                {
+                    let balances = dids
+                        .into_iter()
+                        .take(100) // max query size limit
```
Comment (fmiguelgarcia) [inline pallets/runtime/common/src/runtime.rs:856]: What about using a constant here? Something like: MAX_DID_COUNT or MAX_QUERY_SIZE_LIMIT (from your comment)

File: pallets/runtime/common/src/runtime.rs [hunk 3]
```diff
@@ -845,6 +845,19 @@ macro_rules! runtime_apis {
...
+                        .map(|did| Asset::get_balance_at(ticker, did, checkpoint).into())
+                        .collect();
+                    Ok(balances)
```
Comment (fmiguelgarcia) [inline pallets/runtime/common/src/runtime.rs:859]: It  looks like an `infallible` method, Do we really need to return a `Result<_>`?

**Review by fmiguelgarcia — APPROVED** [review]

File: pallets/runtime/common/src/runtime.rs (hunk 3 above)
Comment (vkomenda) [reply pallets/runtime/common/src/runtime.rs:859]: `rpc_forward_call!` macro requires the return type to be a `Result`. It's either an `Ok` here or in `rpc/src/asset.rs`. So I just made a design decision and placed it here.

**Review by maxsam4 — APPROVED** [review]

**Pull request merged**